### PR TITLE
SONAR-9871 call project webhook on issue change on short branches

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/settings/ChildSettings.java [hunk 1]
```diff
@@ -17,54 +17,67 @@
...
-import org.sonar.api.ce.ComputeEngineSide;
+import org.sonar.api.config.Configuration;
 import org.sonar.api.config.Settings;
+import org.sonar.api.config.internal.ConfigurationBridge;
 
 import static java.util.Objects.requireNonNull;
 
-@ComputeEngineSide
-public class ProjectSettings extends Settings {
+public class ChildSettings extends Settings {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/settings/ChildSettings.java]: MINOR Remove this use of "Settings"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: server/sonar-server/src/main/java/org/sonar/server/settings/ChildSettings.java [hunk 2]
```diff
@@ -17,54 +17,67 @@
...
 
-  private final Map<String, String> projectProps = new HashMap<>();
-  private final Settings globalSettings;
+  private final Settings parentSettings;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/settings/ChildSettings.java]: MINOR Remove this use of "Settings"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: server/sonar-server/src/main/java/org/sonar/server/settings/ChildSettings.java [hunk 3]
```diff
@@ -17,54 +17,67 @@
...
+  private final Map<String, String> localProperties = new HashMap<>();
 
-  public ProjectSettings(Settings globalSettings) {
-    super(globalSettings.getDefinitions(), globalSettings.getEncryption());
-    this.globalSettings = globalSettings;
+  public ChildSettings(Settings parentSettings) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/settings/ChildSettings.java]: MINOR Remove this use of "Settings"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: server/sonar-server/src/main/java/org/sonar/server/settings/ChildSettings.java [hunk 4]
```diff
@@ -17,54 +17,67 @@
...
+    super(parentSettings.getDefinitions(), parentSettings.getEncryption());
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/settings/ChildSettings.java]: MINOR Remove this use of "Settings"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: server/sonar-server/src/main/java/org/sonar/server/settings/ProjectConfigurationLoaderImpl.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import org.sonar.api.config.Settings;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.component.ComponentDto;
+import org.sonar.db.property.PropertyDto;
+
+import static org.sonar.core.util.stream.MoreCollectors.uniqueIndex;
+
+public class ProjectConfigurationLoaderImpl implements ProjectConfigurationLoader {
+  private final Settings globalSettings;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/settings/ProjectConfigurationLoaderImpl.java]: MINOR Remove this use of "Settings"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: server/sonar-server/src/main/java/org/sonar/server/settings/ProjectConfigurationLoaderImpl.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  private final DbClient dbClient;
+
+  public ProjectConfigurationLoaderImpl(Settings globalSettings, DbClient dbClient) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/settings/ProjectConfigurationLoaderImpl.java]: MINOR Remove this use of "Settings"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 issues
* ![MINOR][MINOR] 6 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: server/sonar-server/src/main/java/org/sonar/server/issue/webhook/IssueChangeWebhookImpl.java [hunk 7]
```diff
@@ -107,10 +107,6 @@ private static boolean isMeaningfulTransition(String transitionKey) {
   }
 
   private void callWebHook(IssueChangeData issueChangeData) {
-    if (!webhooks.isEnabled(configuration)) {
-      return;
-    }
-
     Set<String> componentUuids = issueChangeData.getIssues().stream()
```
Comment (ehartmann) [inline server/sonar-server/src/main/java/org/sonar/server/issue/webhook/IssueChangeWebhookImpl.java]: This should be named projectUuids, isn't it ?

**Pull request merged**